python/spark_sklearn/grid_search.py

```diff
@@ -180,35 +240,40 @@ def fit(self, X, y=None):
         """
         return self._fit(X, y, ParameterGrid(self.param_grid))
 
-    def _fit(self, X, y, parameter_iterable):
-        """Actual fitting,  performing the search over parameters."""
+    
+    def fit(self, X, y=None, groups=None, **fit_params):
+
+      if self.fit_params is not None:
```

Comment:
It looks like there are indentation issues here which would make this function do nothing if fit_params is None